ivy/functional/frontends/paddle/tensor/tensor.py

```diff
@@ -908,3 +908,51 @@ def gather(self, y, name=None):
     def gather_(self, y, name=None):
         res = self.gather(self, y)
         return ivy.inplace_update(self, res)
+
+    def _fill_diagonal_tensor_impl(self, x, y, offset=0, dim1=0, dim2=1, inplace=False):
+        inshape = x.shape
+        print(inshape)
+        assert dim1 < len(inshape) and dim1 >= -len(
+            inshape
+        ), "dim1 should be between [-rank, rank) in fill_diagonal_tensor_"
+        assert dim2 < len(inshape) and dim2 >= -len(
+            inshape
+        ), "dim2 should be between [-rank, rank) in fill_diagonal_tensor_"
+        assert len(inshape) >= 2, "Tensor dims should be >= 2 in fill_diagonal_tensor_"
+        dim1 %= len(inshape)
+        dim2 %= len(inshape)
+
+        predshape = []
+        for i in range(len(inshape)):
+            if i != dim1 and i != dim2:
+                predshape.append(inshape[i])
+        diaglen = min(
+            min(inshape[dim1], inshape[dim1] + offset),
+            min(inshape[dim2], inshape[dim2] - offset),
+        )
+        predshape.append(diaglen)
+        assert tuple(predshape) == tuple(y.shape), "the y shape should be {}".format(
+            predshape
+        )
+        if len(y.shape) == 1:
+            y = y.reshape([1, -1])
+
+        diag_start = max(0, offset)
+        diag_end = diag_start + diaglen
+        # should use ivy.inplace_update(self, res) instead
+        if inplace:
+            # Modify the input tensor 'x' directly
+            x[:diaglen, diag_start:diag_end] = y
+            return x
+        else:
+            # Create a copy of 'x' and apply the modification to the copy
+            result = x.clone()
+            result[:diaglen, diag_start:diag_end] = y
+            return result
+
+    # Paddle source code implementation
+    def fill_diagonal_tensor_(self, y, offset=0, dim1=0, dim2=1, name=None):
+        self = self.fill_diagonal_tensor_impl(
```

Comment:
I'm not really sure if assigning `self` is valid (or a good practice) in python, you are basically assigning memory address by itself.